spec/unit/type/maillist_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:maillist) do
+  let(:maillist) do
+    described_class.new(
+      name: 'dummy',
+      provider: :mailman,
+    )
+  end
+
+  let(:provider) { maillist.provider }
+  let(:catalog) { Puppet::Resource::Catalog.new }
+
+  # Mail List aliases are careful not to stomp on managed Mail Alias aliases
+  it 'generates aliases unless they already exist' do
+    allow(maillist).to receive(:catalog) { catalog }
+
+    # test1 is an unmanaged alias from /etc/aliases
+    allow(Puppet::Type.type(:mailalias).provider(:aliases)).to receive(:target_object) { StringIO.new("test1: root\n") }
```

Comment:
It looks like there some dependency between the `mailalias` and `maillist` modules, but I bet this passes before the `mailalias` type is still in puppet proper. Should we put `mailalias` and `maillist` into one module, or have `maillist` depend on `mailalias`?

Reply:
if I bundle install and remove lib/puppet/type/mail* from the installed puppet gem, then the tests fail as expected since they type can't be resolved:

```
$ bx rspec spec

Puppet::Type::Maillist
  generates aliases unless they already exist (FAILED - 1)

Failures:

  1) Puppet::Type::Maillist generates aliases unless they already exist
     Failure/Error: allow(Puppet::Type.type(:mailalias).provider(:aliases)).to receive(:target_object) { StringIO.new("test1: root\n") }

     NoMethodError:
       undefined method `provider' for nil:NilClass
     # ./spec/unit/type/maillist_spec.rb:19:in `block (2 levels) in <top (required)>'
```

It seems like these resources are managed using different command/packages (sendmail vs mailman), so I'm guessing best to keep them as separate modules and add a dependency between them? Thoughts @jhelwig?

Reply:
Yeah, I think it makes sense to keep them separate, and have maillist depend on mailalias, as you can use mailalias without maillist, but not the other way around. 👍